### DISCUSSION TITLE
Enrollment token always invalid

### DISCUSSION THREAD

**Discussion by amjelnes** [DefGuard/defguard#747] [Q&A]
First of all thanks for a great project.

I've managed to install and setup up package based defguard on fully patched Debian 12, I can access main WebUI and Enrollment WebUI so far so good.
Email works fine and enrollment email works, but when clicking the link in the email it does not automatically insert the token value in the enrollment window.
If i then manually copy paste the token i always get "Field is invalid".
Also every time the defguard service is restarted a new token needs to be inserted in the gateway.toml file and gateway service needs to be restarted. Otherwise Gateway will show as disconnected.

What am I doing wrong here?

**Comment by teon** [comment]
@amjelnes thank you - I see that actually you are right - there is a bug/regression in the enrollment WEB UI. Sorry about that!

In a "normal" use-case enrollment is best done through the Desktop Client as it does both: enrollment & configures the client automatically.

Regarding the second issue (restarting core and new token) I don't see this case. Could you provide us with error logs from your gateway?

**Reply by amjelnes** [reply]
Sorry seems I had messed things up a bit .... I've been struggling a bit with certificates and ports.

It seems the problem went away.... not sure what I did to make that happen but think we can write it of as a misconf on my behalf.

**Comment by teon** [comment]
@amjelnes here is the issue for the invalid token:

https://github.com/DefGuard/proxy/issues/80

you can also see when the fixed proxy release will be ready:

https://github.com/DefGuard/proxy/issues/79

**Reply by amjelnes** [reply]
Hi teon,
Not to stress or anything, I know this is probably on a sparetime basis and I appreciate and respect your effort, but are there any time frame for a fixed proxy release?
